encoder/vaapiencoder_h264.cpp

```diff
@@ -912,38 +916,32 @@ Encode_Status VaapiEncoderH264::reorder(const SurfacePtr& surface, uint64_t time
     if (!surface)
         return ENCODE_INVALID_PARAMS;
 
-    ++m_curPresentIndex;
     PicturePtr picture(new VaapiEncPictureH264(m_context, surface, timeStamp));
-    picture->m_poc = ((m_curPresentIndex * 2) % m_maxPicOrderCnt);
 
     bool isIdr = (m_frameIndex == 0 ||m_frameIndex >= keyFramePeriod() || forceKeyFrame);
 
-    /* check key frames */
-    if (isIdr || (m_frameIndex % intraPeriod() == 0)) {
-        ++m_curFrameNum;
-        ++m_frameIndex;
-        /* b frame enabled,  check queue of reorder_frame_list */
-        if (m_numBFrames
```

Comment:
we need consider one corner case. suppose we have n b frame and next frame is idr. We need change nth b frame to P frame and push it to front of reorder list.